backend/typescript/rest/authRoutes.ts

```diff
@@ -103,6 +103,33 @@ authRouter.post("/register", registerRequestValidator, async (req, res) => {
   }
 });
 
+/**
+ * Sends the user an email to reset their password. Used when the user has forgotten their password.
+ * @returns bool - true if successfully sent the email, false otherwise
+ */
+authRouter.post("/forgotPassword", async (req, res) => {
+  try {
+    // Find the user record with that email. If none found, error will be caught
+    const firebaseUserUid = await authService.getFirebaseUserIdByEmail(
+      req.body.email,
+    );
+
+    // Set the user's password to the random access code. This is needed on the front end to verify the user
+    const accessCode = await authService.setTemporaryUserPassword(
+      firebaseUserUid,
+    );
+
+    const currentUser = await userService.getUserByEmail(req.body.email);
+
+    // Send email to user to reset their password
+    await authService.sendPasswordSetupLink(currentUser, accessCode, false);
+
+    res.status(200).json();
+  } catch (error: unknown) {
+    sendErrorResponse(error, res);
```

Comment:
If the email doesn't exist/is incorrect, this should still probably send a success for security/obfuscation. ie we shouldn't reveal whether an email has a valid account

Reply:
done